app/models/location.rb

```diff
@@ -126,15 +130,26 @@ class Location < AbstractModel
   # Callback whenever new version is created.
   versioned_class.before_save do |ver|
     ver.user_id = User.current_id || User.admin_id
+    count_versions = Location.arel_select_count_location_versions(ver)
+
     if (ver.version != 1) &&
-       Location.connection.select_value(%(
-         SELECT COUNT(*) FROM locations_versions
-         WHERE location_id = #{ver.location_id} AND user_id = #{ver.user_id}
-       )).to_s == "0"
+       Location.connection.select_value(count_versions.to_sql).to_s == "0"
+      # if (ver.version != 1) &&
+      #    Location.connection.select_value(%(
+      #      SELECT COUNT(*) FROM locations_versions
+      #      WHERE location_id = #{ver.location_id} AND user_id = #{ver.user_id}
+      #    )).to_s == "0"
       SiteData.update_contribution(:add, :locations_versions)
```

Comment:
Why can't we just use the ActsAsVersioned relation?

```suggestion
  Location::Version.where(location_id: ver.location_id, user_id: ver.user_id).none?
```

Reply:
(Sorry, I'm still learning how to use the suggestions feature!!)

Reply:
Committing this change. If it's ok i will remove the commented out Arel stuff.